pynng/nng.py

```diff
@@ -908,3 +989,103 @@ def close(self):
         check_err(lib.nng_pipe_close(self.pipe))
         self._closed = True
 
+
+class Message:
+    """
+    Python interface for nng_msg.  See
+    https://nanomsg.github.io/nng/man/tip/nng_msg.5.html
+
+    Messages are immutable.
+
+    Warning:
+
+        Access to the message's underlying data buffer can be accessed with the
+        ``buffer`` attribute.  However, care must be taken not to send a message
+        while a reference to the buffer is still alive; if the buffer is used after
+        a message is sent, a segfault or data corruption will result.
+
+    """
+
+    def __init__(self, data, pipe=None):
+        self._pipe = pipe
+        # NB! There are two ways that a user can free resources that an nng_msg
```

Comment:
This comment might be out of date now?
Re:

>So the only way to send a message is with the send() and asend() methods on the object.

Reply:
Good observation.